pkg/providers/cloudstack/validator_registry.go

```diff
@@ -0,0 +1,41 @@
+package cloudstack
+
+import (
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/networkutils"
+	"github.com/aws/eks-anywhere/pkg/providers/cloudstack/decoder"
+)
+
+// ValidatorRegistry exposes a single method for retrieving the CloudStack validator, and abstracts away how they are injected.
+type ValidatorRegistry interface {
+	Get(execConfig *decoder.CloudStackExecConfig) (*Validator, error)
```

Comment:
I was unable to define a generic Validator interface for mocking because it resulted in an import cycle:

```
# github.com/aws/eks-anywhere/pkg/providers/cloudstack
package github.com/aws/eks-anywhere/pkg/providers/cloudstack
	imports github.com/aws/eks-anywhere/pkg/providers/cloudstack/mocks: import cycle not allowed in test
FAIL	github.com/aws/eks-anywhere/pkg/providers/cloudstack [setup failed]
```

Reply:
Yeah you would need to define the validator interface in this package

Reply:
Ahh I see, yeah, that's another issue of having the tests in the same package as the code, we should move away from that.

In the meantime, you can create the mocks in the cloudstack package, that should bypass the circular dependency.

Reply:
Created https://github.com/aws/eks-anywhere/issues/3921 to address the unit test refactoring work. I don't actually need mocks for this validator currently, so maybe that would be something that could be done when the v2 controller is ready to be reviewed.